reactor-core/src/main/java/reactor/util/context/ContextN.java

```diff
@@ -60,8 +63,38 @@ public Context put(Object key, Object value) {
 	}
 
 	@Override
+	public Context delete(Object key) {
```

Comment:
nit: maybe over optimising? If you have a program that actively manipulates contexts enough to downgrade from ContextN to Context4, it *may* typically be more efficient to stay as a map forever more.

Reply:
ignore... that is probably moot because you copy the entire map on every mutation anyway.  Would be interesting to benchmark with some persistent immutable data structures (is that the right word)?